components/create-collective/NewCreateCollective.js

```diff
@@ -0,0 +1,259 @@
+                      Create an account (or sign in) to start a collective.
+                    </P>
+                  </Box>
+                </Flex>
+                <SignInOrJoinFree />
+              </Flex>
+            </Fragment>
+          )}
+          {canApply && LoggedInUser && !category && (
+            <CollectiveCategoryPicker query={query} onChange={(key, value) => this.handleChange(key, value)} />
+          )}
+          {canApply && LoggedInUser && category && category !== 'opensource' && (
+            <CreateCollectiveForm
+              host={this.props.host}
+              collective={this.state.collective}
+              onSubmit={this.createCollective}
+              onChange={(key, value) => this.handleChange(key, value)}
+              error={error}
+              query={query}
+            />
+          )}
+          {canApply && LoggedInUser && category === 'opensource' && !form && (
+            <ConnectGithub token={token} query={query} onChange={(key, value) => this.handleChange(key, value)} />
+          )}
+          {canApply && LoggedInUser && category === 'opensource' && form && (
+            <CreateCollectiveForm
+              host={this.props.host}
+              collective={this.state.collective}
+              onSubmit={this.createCollective}
+              onChange={(key, value) => this.handleChange(key, value)}
+              error={error}
+              query={query}
+            />
+          )}
+        </div>
+      </Page>
+    );
+  }
+}
+
+const createCollectiveQueryV2 = gqlV2`
+  mutation createCollective(
+    $collective: CreateCollectiveInput!
+    $host: AccountInput
+    $automateApprovalWithGithub: Boolean
+  ) {
+    createCollective(collective: $collective, host: $host, automateApprovalWithGithub: $automateApprovalWithGithub) {
+      name
+      slug
+      tags
+      description
+      githubHandle
+    }
+  }
+`;
+
+const addCreateCollectiveMutationV2 = graphql(createCollectiveQueryV2, {
+  options: {
+    context: { apiVersion: '2' },
+  },
+  props: ({ mutate }) => ({
+    createCollectiveV2: async ({ collective, host }) => {
+      const CreateCollectiveInputType = pick(collective, [
+        'slug',
+        'name',
+        'description',
+        'githubHandle',
+        'tags',
+        CreateCollectiveInputType,
+      ]);
+      return await mutate({
+        variables: {
+          collective: CreateCollectiveInputType,
+          host: host,
+        },
+        update: (store, { data: { createCollectiveV2 } }) => {
+          const data = store.readQuery({ query: getLoggedInUserQuery });
+          data.LoggedInUser.memberOf.push({
+            __typename: 'Member',
+            collective: createCollectiveV2,
+            role: 'ADMIN',
+          });
+          store.writeQuery({ query: getLoggedInUserQuery, data });
+        },
+      });
+    },
+  }),
+});
+
+export default compose(addCreateCollectiveMutationV2)(injectIntl(withUser(NewCreateCollective)));
```

Comment:
You don't need compose, just `addCreateCollectiveMutationV2(` if I'm not mistaken.